hq_superset/utils.py

```diff
@@ -37,3 +47,51 @@ def get_role_name_for_domain(domain):
     # Prefix in-case domain name matches with known role names such as admin
     # Same prefix pattern as schema only by coincidence, not a must.
     return f"{DOMAIN_PREFIX}{domain}"
+
+
+def get_column_dtypes(datasource_defn):
+    """
+    Maps UCR column data types to Pandas data types.
+
+    See corehq/apps/userreports/datatypes.py for possible data types.
+    """
+    # TODO: How are array indicators handled in CSV export?
+    pandas_dtypes = {
+        'date': 'datetime64[ns]',
+        'datetime': 'datetime64[ns]',
+        'string': 'string',
+        'integer': 'int64',
+        'decimal': 'float64',
+        'small_integer': 'int8',  # TODO: Is this true?
```

Comment:
From my understanding, this maps to SQLAlchemy's [SmallInteger](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/sql/columns.py#L52), which SQLAlchemy [maps](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.SmallInteger) to Postgres's SMALLINT which is of [2 bytes](https://www.postgresql.org/docs/current/datatype-numeric.html).

Reply:
These aren't SQLAlchemy datatypes. They are Pandas [Dataframe datatypes](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dtypes.html), which we use [here, when reading the CSV](https://github.com/dimagi/hq_superset/pull/13/commits/63c22451dbb446500f325da37a65e5013eae4d23#diff-904cccd73cc4407182ae5580289ad118ed1be682677edf1c7e0ba03489788699R129).

If "small_integer" in UCRs is 2 bytes (-32,768 - 32,767) we should be fine. (I suspect we use it for booleans as well, because `'boolean'` [is not included in DATA_TYPE_CHOICES](https://github.com/dimagi/commcare-hq/blob/cc93bc6ab718bde08e8d16e374d5cc25eff94958/corehq/apps/userreports/datatypes.py#L9). (?!) ) I just don't want to be in a situation where 8 bytes is not enough.